tiflash: update tiflash deploy using tiup (#2194)
pingcap/docs: reference/tiflash/deploy.md

### Fresh TiFlash deployment

> TiUP Cluster is the deployment tool for TiDB 4.0 or later versions. It is recommended that you use TiUP Cluster to install and deploy TiFlash. The steps are as follows:

Reviewer: ```suggestion
TiUP cluster is the deployment tool for TiDB 4.0 or later versions. It is recommended that you use TiUP cluster to install and deploy TiFlash. The steps are as follows:
```
TiUP deployment doc uses `TiUP cluster`. Please update this word in other places (if any). @ran-huang

> 1. [Install TiUP](/how-to/deploy/orchestrated/tiup.md).

Reviewer: ```suggestion
1. [Install TiUP](/how-to/deploy/orchestrated/tiup.md#step-2-install-tiup-on-the-control-machine).
```

2. Install the TiUP cluster component.

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster
    ```

3. Write the topology configuration file and save it as `topology.yaml`.

>     For details, see [the topology example](https://github.com/pingcap-incubator/tiops/blob/master/topology.example.yaml).

Reviewer: ```suggestion
    You can refer to [the topology configuration file template](https://github.com/pingcap-incubator/tiops/blob/master/topology.example.yaml).
```

>     In addition to configuring the TiDB cluster, you also need to configure the IP of tiflash servers in `tiflash_servers`. Currently the configuration only supports IP but not domain name.

Reviewer: ```suggestion
    In addition to configuring the TiDB cluster, you also need to configure the IP of TiFlash servers in `tiflash_servers`. Currently the configuration only supports IP but not domain name.
```

    {{< copyable "" >}}

    ```ini
    tiflash_servers:
      - host: 172.19.0.103
        data_dir: /data1/tiflash/data,/data2/tiflash/data
    ```

4. Refer to the TiUP deployment process, and complete the following steps:

    * Deploy the TiDB cluster (`test` is the cluster name):
    
        {{< copyable "shell-regular" >}}

        ```shell
        tiup cluster deploy test v4.0.0-rc topology.yaml  -i ~/.ssh/id_rsa
        ```
    
    * Start the TiDB cluster:

        {{< copyable "shell-regular" >}}

        ```shell
        tiup cluster start test
        ```

5. View the cluster status:

    {{< copyable "shell-regular" >}}

    ```shell
    tiup cluster display test
    ``` 

### Add TiFlash component to an existing TiDB cluster

1. First, confirm that your current TiDB version supports TiFlash; otherwise, you need to upgrade your TiDB cluster to 4.0 rc or higher versions.

2. Execute the `config set enable-placement-rules true` command in [pd-ctl](/reference/tools/pd-control.md) (`resources/bin` in the tidb-ansible directory includes the pd-ctl binary file) to enable PD's Placement Rules feature.

>     Currently, pd-ctl is not connected to TiUP Cluster, you need to [download pd-ctl manually](https://download.pingcap.org/tidb-v4.0.0-rc-linux-amd64.tar.gz).

Reviewer: ```suggestion
    Currently, pd-ctl is not connected to TiUP cluster, you need to [manually download pd-ctl](https://download.pingcap.org/tidb-v4.0.0-rc-linux-amd64.tar.gz).
```

> 3. Refer to [Scale out a TiFlash node](reference/tiflash/scale.md#scale-out-a-tiflash-node) and deploy TiFlash.

Reviewer: ```suggestion
3. Refer to [Scale out a TiFlash node](/reference/tiflash/scale.md#scale-out-a-tiflash-node) and deploy TiFlash.
```
